docs-client/src/containers/MethodPage/DebugPage.tsx

```diff
@@ -237,14 +276,86 @@ class DebugPage extends React.PureComponent<Props, State> {
 
   private onCopy = () => {
     const response = this.state.debugResponse;
-    const textArea = document.createElement('textarea');
-    textArea.style.opacity = '0.0';
-    textArea.value = response;
-    document.body.appendChild(textArea);
-    textArea.focus();
-    textArea.select();
-    document.execCommand('copy');
-    document.body.removeChild(textArea);
+    if (response.length > 0) {
+      DebugPage.copyTextToClipboard(response);
+      this.showSnackbar('The response has been copied to the clipboard.');
+    }
+  };
+
+  private onExport = () => {
+    const escapeSingleQuote = (text: string) => text.replace(/'/g, `'\\''`);
+
+    const endpointPath = this.state.endpointPath; // annotated service only
+    const additionalHeaders = this.state.additionalHeaders;
+    const queries = this.state.additionalQueries;
+    const requestBody = this.state.requestBody;
+
+    try {
+      if (this.props.useRequestBody) {
+        DebugPage.validateJsonObject(requestBody, 'request body');
+      }
+
+      if (additionalHeaders) {
+        DebugPage.validateJsonObject(additionalHeaders, 'headers');
+      }
+
+      const headers =
+        (additionalHeaders && JSON.parse(additionalHeaders)) || {};
+
+      // window.location.origin may have compatibility issue
+      // https://developer.mozilla.org/en-US/docs/Web/API/Window/location#Browser_compatibility
+      const host =
+        `${window.location.protocol}//${window.location.hostname}` +
+        `${window.location.port ? `:${window.location.port}` : ''}`;
+
+      const method = this.props.method;
+      const transport = TRANSPORTS.getDebugTransport(method);
+      if (!transport) {
+        throw new Error("This method doesn't have a debug transport.");
+      }
+
+      const httpMethod = method.httpMethod;
+      const [path, body] = transport.curlPathAndBody(
+        method,
+        escapeSingleQuote(requestBody),
+      );
+      let uri;
+
+      if (this.props.isAnnotatedHttpService) {
+        if (this.props.exactPathMapping) {
+          uri =
+            `'${host}${escapeSingleQuote(path)}` +
+            `${queries.length > 0 ? `?${escapeSingleQuote(queries)}` : ''}'`;
+        } else {
+          this.validateEndpointPath(endpointPath);
+          uri = `'${host}${escapeSingleQuote(endpointPath)}'`;
+        }
+      } else {
+        uri = `'${host}${escapeSingleQuote(path)}'`;
+      }
+
+      headers['content-type'] = transport.getDebugMimeType();
+      if (process.env.WEBPACK_SERVE === 'true') {
+        headers[docServiceDebug] = 'true';
+      }
+
+      const header = Object.keys(headers)
+        .map((title) => {
```

Comment:
nit: How about renaming `title` to `name` for better readability?

Reply:
Changed to `name`.